### PR TITLE
optimize KuduRowDeserializer use RowTypeInfo in runtime

### PR DESCRIPTION
**Pull request by beyond-up**
Signed-off-by:

## Pre-Checklist

Note: Please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/bytedance/bitsail/blob/master/docs/contributing.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests.

## Purpose

Use TypeInfo in runtime to parse data in KuduRowDeserializer

## Related Issues

https://github.com/bytedance/bitsail/issues/258

## New Behavior (screenshots if needed)

N/A

### CODE REVIEW

File: bitsail-common/src/main/java/com/bytedance/bitsail/common/typeinfo/TypeInfos.java [hunk 1]
```diff
@@ -64,6 +64,8 @@ public final class TypeInfos {
       new BasicTypeInfo<>(LocalTime.class);
   public static final TypeInfo<LocalDateTime> LOCAL_DATE_TIME_TYPE_INFO =
       new BasicTypeInfo<>(LocalDateTime.class);
+  public static final TypeInfo<byte[]> BINARY_TYPE_INFO =
```
Comment (BlockLiu) [inline bitsail-common/src/main/java/com/bytedance/bitsail/common/typeinfo/TypeInfos.java:67]: Actually we have `binary` type already.
You can directly use `BasicArrayTypeInfo.BINARY_TYPE_INFO`.
So I suggest this one can be removed.

File: bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/reader/KuduRowDeserializer.java [hunk 2]
```diff
@@ -48,48 +52,53 @@ public Row convert(RowResult rowResult) {
...
+  private Function<RowResult, Object> initWrappedConverter(String columnName, TypeInfo<?> typeInfo) {
+    Function<RowResult, Object> converter = initConverter(columnName, typeInfo);
     return rowResult -> rowResult.isNull(columnName) ? null : converter.apply(rowResult);
   }
 
-  private Function<RowResult, Object> initConverter(ColumnInfo columnInfo) {
-    String columnName = columnInfo.getName();
-    String typeName = columnInfo.getType().trim().toUpperCase();
+  private Function<RowResult, Object> initConverter(String columnName, TypeInfo<?> typeInfo) {
+    if (!(typeInfo instanceof BasicTypeInfo)) {
```
Comment (BlockLiu) [inline bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/reader/KuduRowDeserializer.java:61]: BINARY_TYPE_INFO is not instance of BasicTypeInfo.

File: bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/reader/KuduRowDeserializer.java [hunk 3]
```diff
@@ -48,48 +52,53 @@ public Row convert(RowResult rowResult) {
...
+    }
+    if (TypeInfos.BIG_DECIMAL_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getDecimal(columnName);
+    }
+    if (TypeInfos.STRING_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getString(columnName);
+    }
+    if (BasicArrayTypeInfo.BINARY_TYPE_INFO.getTypeClass() == curClass) {
+      return rowResult -> rowResult.getBinary(columnName);
     }
```
Comment (BlockLiu) [inline bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/reader/KuduRowDeserializer.java:101]: getBinary() method will return a `ByteBuffer` object. 
It's supposed to return a `byte[]` object.

Comment (beyond-up) [reply bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/reader/KuduRowDeserializer.java:96]: > getBinary() method will return a `ByteBuffer` object. It's supposed to return a `byte[]` object.

Thank you for you suggestion , I have fixed, please help me review again thx ! @BlockLiu

**Comment by BlockLiu** [comment]
LGTM

**Pull request merged**